core/environment.ts

```diff
@@ -151,12 +151,13 @@ export class Environment {
         return __exports;
       `);
       code = `
-        return new Function(
+        return new (async function(){}).constructor(
```

Comment:
Should this constructor be cached somewhere?
I mean, `const asynConstructor = (async function(){}).constructor;` and use the variable later. Maybe it has any performance benefit.

Reply:
We can do that, but the scopes are a bit awkward (it would need to be passed as argument into the template `constructor` function) and it's probably slightly faster but I didn't see significant differences locally. Sometimes it's faster, other times it's vice versa. I feel like it'd be a shame to ruin the elegance of the signatures `constructor(__env)` and `__template(it)` :stuck_out_tongue:

Reply:
Ah, I didn't notice that this is actually a string to evaluate later, not real code.
Yeah, I agree.